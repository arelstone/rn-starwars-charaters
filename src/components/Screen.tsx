import React, { FC, ReactNode } from 'react';
import { StyleSheet, View } from 'react-native';
import { SafeAreaViewProps } from 'react-native-safe-area-context';

interface ScreenProps extends SafeAreaViewProps {
    children: ReactNode;
}

const Screen: FC<ScreenProps> = ({ children, ...props }) => {
    const style = [styles.container, props.style || {}];

    return <View
        {...props}
        {...{ style }}
    >
        {children}
    </View>;
};

export default Screen;

const styles = StyleSheet.create({
    container: {
        flex: 1,
    },
});
