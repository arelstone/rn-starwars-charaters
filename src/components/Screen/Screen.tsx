import React, { FC, ReactNode } from 'react';
import { SafeAreaView, StyleSheet } from 'react-native';
import { SafeAreaViewProps } from 'react-native-safe-area-context';

interface ScreenProps extends SafeAreaViewProps {
    children: ReactNode;
}

const Screen: FC<ScreenProps> = ({ children, ...props }) => {
    const style = [styles.container, props.style || {}];

    return <SafeAreaView
        {...props}
        {...{ style }}
    >
        {children}
    </SafeAreaView>;
};

export default Screen;

const styles = StyleSheet.create({
    container: {
        flex: 1,
    },
});
