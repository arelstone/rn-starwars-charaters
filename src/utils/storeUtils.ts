export const getNumber = (string: string) => Number(string.replace(/\D/g, ''));

export const nextPage = (response: any): number => {
    return response.next && getNumber(response.next);
};
