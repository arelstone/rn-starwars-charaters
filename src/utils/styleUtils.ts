import { ViewStyle } from 'react-native';

export const BORDER = (borderColor: string = 'red'): ViewStyle => ({
    borderWidth: 1,
    borderColor,
    borderStyle: 'solid',
});
