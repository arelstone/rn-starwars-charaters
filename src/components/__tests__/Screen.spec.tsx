import React from 'react';
import { ShallowWrapper, shallow } from 'enzyme';
import Screen from '../Screen';
import { Text } from 'react-native';
import { findElementByTestId } from '../../utils/testUtils';

let wrapper: ShallowWrapper;

describe('<Screen />', () => {
    beforeAll(() => {
        wrapper = shallow(<Screen>
            <Text testID="Children">Hello world</Text>
        </Screen>);
    });

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should display children', () => {
        expect(findElementByTestId(Text, 'Children', wrapper).prop('children')).toEqual('Hello world');
    });

    it('should apply styles', () => {
        wrapper.setProps({
            style: {
                backgroundColor: 'red',
            },
        });

        expect(wrapper.prop('style')).toEqual(
            expect.arrayContaining([
                expect.objectContaining({ flex: 1 }),
                expect.objectContaining({ backgroundColor: 'red' }),
            ]),
        );
    });
});
