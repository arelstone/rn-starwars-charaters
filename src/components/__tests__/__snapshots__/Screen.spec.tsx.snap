// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Screen /> should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {},
    ]
  }
>
  <Text
    testID="Children"
  >
    Hello world
  </Text>
</View>
`;
